src/logic/Window.swift

```diff
@@ -83,13 +121,13 @@ class Window {
     }
 
     func refreshThumbnail() {
-        guard let cgImage = cgWindowId.screenshot() else { return }
+        guard let cgImage = cgWindowId.screenshot(self) else { return }
         thumbnail = NSImage(cgImage: cgImage, size: NSSize(width: cgImage.width, height: cgImage.height))
         thumbnailFullSize = thumbnail!.size
     }
 
     func close() {
-        if isWindowlessApp { return }
+        if isWindowlessApp || self.axUiElement == nil { return }
```

Comment:
Does this mean we can't close a window if we haven't visited its Space? That not ideal...

Reply:
It's a pity that HyperSwitch also can't do that either.

Reply:
I just tested and it can.

I did this to test:

* Opened a new window
* Send it to another Space
* Launch HyperSwitch
* Press alt-tab
* Notice that HS shows the window
* Hover the mouse over the window in HS
* A close button appears, press it
* The window on the other Space is closer, even though I never navigated to that Space

How can HS close the window without the AX reference?

Reply:
On macOS 12.3.1, I did the steps same to you, but HyperSwitch just took me to that space and didn't close the window. Maybe the private APIs they used also broken.